### PR TITLE
Issue #214 automatically log extra data with a context manager

### PR DESCRIPTION
**Pull request by soxofaan**
Issue #214 proof of concept to automatically log extra data with a context manager

### CODE REVIEW

File: openeo_driver/util/logging.py [hunk 1]
```diff
@@ -345,6 +345,51 @@ def filter(self, record: logging.LogRecord) -> bool:
         return True
 
 
+class ContextBasedExtraInjectingFilter(logging.Filter):
```
Comment (soxofaan) [inline openeo_driver/util/logging.py:348]: maybe just `ExtraLoggingFilter` to keep it short

**Comment by bossie** [comment]
@soxofaan does my test cover the threading behaviour?

**Comment by soxofaan** [comment]
nice, one thing  that might be added to that test is some way to make sure that there is (forced) context switching between the threads, because now a passing test could just be the result of each thread finishing before the next has started

File: openeo_driver/util/logging.py (hunk 1 above)
Comment (soxofaan) [reply openeo_driver/util/logging.py]: done (also leaves open the option to cover non-context-manager use cases)

**Comment by soxofaan** [comment]
rebased and merged in e276c325de6f9e4e8c7e362a70534d61a5407336

**Pull request closed without merging**